test_mutable_primitives.py

```diff
@@ -32,6 +32,13 @@ def test_eq(self):
         assert DEFAULTS[self.name][0] == self.typ(DEFAULTS[self.name][0])
         assert DEFAULTS[self.name][0] != self.typ(DEFAULTS[self.name][1])
 
+        # check boolean evaluation
+        for val in DEFAULTS[self.name]:
```

Comment:
Can you move this out of test_eq and into a test_truthiness (or something less ridiculous)?

Also you might want to add defaults that will eval to False for the non-bool tests.